### ISSUE TITLE
🐛 Biome encountered an unexpected error

### ISSUE THREAD

**Issue by illandril** [biomejs/biome#2728]
### Environment information

```block
CLI:
  Version:                      1.7.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           "v20.12.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.5.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

I get unexpected errors when running `npx biome check .` or `npx biome lint .` (but not `npx biome format .`) in this commit (and the parent commit) of this project: https://github.com/illandril/FoundryVTT-types/commit/cd7cc216f0e1afca5d0a774268b36049fd921a74
```
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates\biome_js_semantic\src\semantic_model\scope.rs:115:33
Thread Name: biome::worker_14
Message: no entry found for key

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates\biome_js_semantic\src\semantic_model\scope.rs:115:33
Thread Name: biome::worker_0
Message: no entry found for key

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates\biome_js_semantic\src\semantic_model\scope.rs:115:33
Thread Name: biome::worker_18
Message: no entry found for key
```

### Expected result

No unexpected errors

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Duplicate of https://github.com/biomejs/biome/issues/2683
